src/core/include/antioch/kinetics_conditions.h

```diff
@@ -0,0 +1,175 @@
+//-----------------------------------------------------------------------bl-
+//--------------------------------------------------------------------------
+//
+// Antioch - A Gas Dynamics Thermochemistry Library
+//
+// Copyright (C) 2014 Paul T. Bauman, Benjamin S. Kirk, Sylvain Plessis,
+//                    Roy H. Stonger
+// Copyright (C) 2013 The PECOS Development Team
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the Version 2.1 GNU Lesser General
+// Public License as published by the Free Software Foundation.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc. 51 Franklin Street, Fifth Floor,
+// Boston, MA  02110-1301  USA
+//
+//-----------------------------------------------------------------------el-
+
+#ifndef ANTIOCH_KINETICS_CONDITIONS_H
+#define ANTIOCH_KINETICS_CONDITIONS_H
+
+// Antioch
+#include "antioch/antioch_asserts.h"
+#include "antioch/particle_flux.h"
+
+// C++
+#include <vector>
+#include <map>
+
+namespace Antioch{
+
+  /*! This class contains the conditions of the chemistry
+
+      Conditions are:
+        - Temperature(s)
+        - particle flux
+        - any other not yet implemented conditions (catalysis for instance)
+
+      Pressure is given by the molecular composition of the mixture,
+      we are in a gaz phase, so the state equation will give pressure (typically
+      ideal gas).
+
+      The idea is to avoid any copy of anything as much as we can, 
+      we store pointers but deal only with references. Nothing
+      belongs to this object, if there is any cleaning to do,
+      it must be done elsewhere.
+
+      Might be interesting to see about the "double" case
+      for temperature: faster to copy instead of copying
+      the reference, and it's expected to be the classic
+      case.
+   */
+  template <typename StateType, 
+            typename VectorStateType = std::vector<StateType> >
+  class KineticsConditions
+  {
+        public:
+
+          KineticsConditions();
```

Comment:
Do we ever want to do this? I'd make this private and force the user to provide a temperature to construct it.